src/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs

```diff
@@ -523,201 +524,347 @@ public void ActivitySpanIdTests()
+            activity.SetParentId("00-xyz3456789abcdef0123456789abcdef-0123456789abcdef-01");
+            activity.Start();
+            Assert.Equal(ActivityIdFormat.W3C, activity.IdFormat);
+            Assert.True(IdIsW3CFormat(activity.Id));
+        }
 
-                // non hex chars are not supported in parentSpanId
-                activity = new Activity("activity10");
-                activity.SetParentId("00-0123456789abcdef0123456789abcdef-x123456789abcdef-01");
-                activity.Start();
-                Assert.Equal(ActivityIdFormat.W3C, activity.IdFormat);
-                Assert.True(IdIsW3CFormat(activity.Id));
-                Assert.Equal("0000000000000000", activity.ParentSpanId.ToHexString());
-                Assert.Equal("0123456789abcdef0123456789abcdef", activity.TraceId.ToHexString());
-                activity.Stop();
+        [Fact]
+        public void ParentSpanId_W3CNonHexCharsNotSupportedAndDoesNotThrow()
+        {
+            Activity activity = new Activity("activity10");
+            activity.SetParentId("00-0123456789abcdef0123456789abcdef-x123456789abcdef-01");
+            activity.Start();
+            Assert.Equal(ActivityIdFormat.W3C, activity.IdFormat);
+            Assert.True(IdIsW3CFormat(activity.Id));
+            Assert.Equal("0000000000000000", activity.ParentSpanId.ToHexString());
+            Assert.Equal("0123456789abcdef0123456789abcdef", activity.TraceId.ToHexString());
+        }
 
-                // ParentSpanId from parent Activity
+        [Fact]
+        public void IdFormat_W3CForcedOveeridesParentActivityIdFormat()
+        {
+            RemoteExecutor.Invoke(() =>
+            {
                 Activity.DefaultIdFormat = ActivityIdFormat.W3C;
                 Activity.ForceDefaultIdFormat = true;
 
-                parent = new Activity("parent").Start();
-                activity = new Activity("parent").Start();
-                Assert.Equal(parent.SpanId.ToHexString(), activity.ParentSpanId.ToHexString());
+                Activity parent = new Activity("parent").Start();
+                Activity activity = new Activity("child").Start();
+                Assert.Equal(parent.SpanId.ToHexString(), activity.ParentSpanId.ToHexString()); ;
+            }).Dispose();
+        }
 
-                activity.Stop();
-                parent.Stop();
-            }
-            finally
+        [Fact]
+        public void SetIdFormat_CanSetHierarchicalBeforeStart()
+        {
+            Activity activity = new Activity("activity1");
+            Assert.Equal(ActivityIdFormat.Unknown, activity.IdFormat);
+            activity.SetIdFormat(ActivityIdFormat.Hierarchical);
+            Assert.Equal(ActivityIdFormat.Hierarchical, activity.IdFormat);
+            activity.Start();
+            Assert.Equal(ActivityIdFormat.Hierarchical, activity.IdFormat);
+
+            // cannot change after activity starts
+            activity.SetIdFormat(ActivityIdFormat.W3C);
```

Comment:
Trying to change it after it's started is a nop rather than an error?

Reply:
yes, this is default Activity behavior: swallow monitoring errors rather than breaking applications. Errors are still accessible as first chance exceptions.  The reason is that user code does not normally control instrumentation with Activities and won't be able to fix it.